tests/test_arrow_writer.py

```diff
@@ -89,6 +90,45 @@ def test_write(fields, writer_batch_size):
     _check_output(output.getvalue(), expected_num_chunks=num_examples if writer_batch_size == 1 else 1)
 
 
+@pytest.mark.parametrize("writer_batch_size", [None, 1, 10])
+@pytest.mark.parametrize(
+    "fields", [None, {"col_1": pa.string(), "col_2": pa.int64()}, {"col_1": pa.string(), "col_2": pa.int32()}]
+)
+def test_key_datatype(fields, writer_batch_size):
+    output = pa.BufferOutputStream()
+    schema = pa.schema(fields) if fields else None
+    with ArrowWriter(
+        stream=output,
+        schema=schema,
+        writer_batch_size=writer_batch_size,
+        hash_salt="split_name",
+        check_duplicates=True,
+    ) as writer:
+        with pytest.raises(InvalidKeyError):
```

Comment:
Can you also add a test that makes sure that finalizing after using keys doesn't raise errors ?
The only tests involving keys are for checking errors to we're just missing the one that checks the case where evrything is fine :)